### DISCUSSION TITLE
PDF to Images: How to start numbering file names at 1 instead of 0?

### DISCUSSION THREAD

**Discussion by johnmapeson** [ImageMagick/ImageMagick#7492] [Help] [answered]
### ImageMagick version

7.1.1-34

### Operating system, version and so on

macOS Sonoma 14.5

### Description

To convert PDF to images, I can use, for example,

```
magick -units PixelsPerInch -density 300 input.pdf -background white -alpha remove -alpha off output%d.png
```

But the files are numbered from `0`, not from `1`: `output0.png`, `output1.png`, and so on. Is it possible to change it, so that they will be named from `1` instead? (`output1.png`, `output2.png`, and so on.)

**Comment by fmw42** [comment, marked as answer]
Add -scene 1 to your command after the input and before the output.

**Reply by johnmapeson** [reply]
Thanks. And maybe you know, is there a syntax to add as many leading zeros as necessary? If I use, for example, `...output%04d.png` in the command above, the filenames are numbered like `0001`, `0021`, etc. That is, the number `4` is hard-coded. But what I would prefer instead is that if the input PDF have 1-9 pages, the output images should not have leading zeros; if the input PDF have 10-99 pages, the output images should be numbered like `01`, `21`, etc.; if the input PDF  have 100-999 pages, the output images should be numbered like `001`, `021`, `321`, etc. And so on. Is it possible somehow?

**Comment by snibgo** [comment]
>  Is it possible somehow?

Not within a single ImageMagick command. You could first count the images, and use an environment variable to use `02` or `03` etc.
